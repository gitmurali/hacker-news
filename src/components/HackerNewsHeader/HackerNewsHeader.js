import React, { Fragment } from 'react'

import HackerNewsLogo from '@resources/images/y18.gif'

import styles from './HackerNewsHeader.scss'

const HackerNewsHeader = () => {
  return (
    <Fragment>
      <header>
        <h1>Hacker News</h1>
      </header>
      <div className={styles.header}>
        <img src={HackerNewsLogo} className={styles.logo} />
        <a href='#'>
          <span>top</span>
        </a>
        <a href='#'>
          <span>|</span>
        </a>
        <a href='#'>
          <span>new</span>
        </a>
      </div>
    </Fragment>
  )
}

export default HackerNewsHeader
