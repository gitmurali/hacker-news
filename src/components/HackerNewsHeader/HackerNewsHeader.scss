.header {
  background: $orange;
  padding: 5px;
  display: flex;
  align-items: center;
  gap: 5px;
  color: $white;
  font-size: 12px;
}

.logo {
  border: 1px solid $white;
}
