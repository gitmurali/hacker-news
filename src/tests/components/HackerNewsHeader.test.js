import React from 'react'

import HackerNewsHeader from '@components/HackerNewsHeader'
import { shallow, mount } from 'enzyme'
import { IntlProvider } from 'react-intl'

describe('Component:HackerNewsHeader', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(
      <IntlProvider locale='en'>
        <HackerNewsHeader />
      </IntlProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('should render header', () => {
    const wrapper = mount(
      <IntlProvider locale='en'>
        <HackerNewsHeader />
      </IntlProvider>
    )
    expect(wrapper.find('a')).toHaveLength(3)
  })
})
