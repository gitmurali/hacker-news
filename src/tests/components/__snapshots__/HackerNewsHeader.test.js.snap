// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:HackerNewsHeader should match snapshot 1`] = `<HackerNewsHeader />`;
